pkg/vpcmodel/connectivityAbstraction.go

```diff
@@ -22,19 +24,33 @@ import (
 //      nodeSetToNodeSet: connections of   <node     in the nodeSet>  ->  <node     in the nodeSet>
 //      otherFromNodeSet: connections of   <node     in the nodeSet>  ->  <node not in the nodeSet>
 //      otherToNodeSet:   connections of   <node not in the nodeSet>  ->  <node     in the nodeSet>
-// 2. for the last three groups, we verify that the abstraction assumption holds
-//     todo: this check is not complete in code yet, and currently we ignore its result
-// 3. we do the abstraction (for this PR, even if the abstraction assumption does not hold):
+// 2. we do the abstraction on the last three groups:
 //    the connectivity of n->NS is union of all of n->n1, n->n2, n->n3
-//    todo: what to do if the abstraction assumption does not hold?
-
-//nolint:gocritic // for now, we comment out the check calls
-func nodeSetConnectivityAbstraction(nodesConn GeneralConnectivityMap, nodeSet NodeSet) GeneralConnectivityMap {
-	otherToOther, nodeSetToNodeSet, otherFromNodeSet, otherToNodeSet := partitionConnectivityByNodeSet(nodesConn, nodeSet)
-	// checkConnectivityAbstractionValidity(otherFromNodeSet, nodeSet, false)
-	// checkConnectivityAbstractionValidity(otherToNodeSet, nodeSet, true)
-	// checkConnectivityAbstractionValidity(nodeSetToNodeSet, nodeSet, true)
-	return mergeConnectivityWithNodeSetAbstraction(otherToOther, nodeSetToNodeSet, otherFromNodeSet, otherToNodeSet, nodeSet)
+// 3. for the last three groups, we collect the missing connections.
+// 4. we add the 4th group to connectivity
+
+// NodeSetAbstraction abstract nodesets, one after the other
```

Comment:
NodeSetAbstraction is more general - it can work not only on load balancer, but any nodeset that we have.
I didnt made it LB specific so it might be useful later.
this file does not have the words Load balancer in it